lib/RestructureCFG/CycleEquivalenceAnalysis.cpp

```diff
@@ -0,0 +1,468 @@
+
+// Debug logger.
+Logger<> CycleEquivalenceAnalysisLogger("cycle-equivalence");
+
+template<class GraphT, class GT>
+void CycleEquivalenceAnalysis<GraphT, GT>::insertEdge(EdgeDescriptor E,
+                                                      BracketDescriptor BD) {
+
+  // We marshal the `EdgeDescriptor` object to unwrap the underlying edge, not
+  // wrapped in any `GenericGraph` node
+  using NakedEdgeDescriptor = CycleEquivalenceClass<NodeT>::EdgeDescriptor;
+  NakedEdgeDescriptor NakedEdge = NakedEdgeDescriptor(E.first->getBlock(),
+                                                      E.second->getBlock());
+
+  auto MapIt = BracketToClass.find(BD);
+  if (MapIt != BracketToClass.end()) {
+    size_t Index = MapIt->second;
+    revng_assert(CycleEquivalenceClasses.size() > Index);
+    auto &EClass = CycleEquivalenceClasses[Index];
+    EClass.insertEdge(NakedEdge);
+  } else {
+    size_t NewIndex = CycleEquivalenceClasses.size();
+    BracketToClass.emplace(BD, NewIndex);
+    ClassToBracketDescriptor.emplace(NewIndex, BD);
+    auto &EClass = CycleEquivalenceClasses.emplace_back(NewIndex);
+    EClass.insertEdge(NakedEdge);
+  }
+}
+
+template<class GraphT, class GT>
+void CycleEquivalenceAnalysis<GraphT, GT>::print() {
+
+  // Print the Brackets Analysis results
+  revng_log(CycleEquivalenceAnalysisLogger, "");
+  revng_log(CycleEquivalenceAnalysisLogger,
+            "Ordered Bracket Analysis Results:");
+  for (auto &EClass : CycleEquivalenceClasses) {
+    EClass.print();
+  }
+
+  // Print the correspondence between the Equivalence Class and the Bracket
+  // Descriptor which gave origin to it
+  revng_log(CycleEquivalenceAnalysisLogger, "");
+  revng_log(CycleEquivalenceAnalysisLogger, "Class Bracket Correspondence:");
+  for (const auto &[Index, BD] : ClassToBracketDescriptor) {
+    auto &Edge = BD.first;
+    auto &Size = BD.second;
+    revng_log(CycleEquivalenceAnalysisLogger,
+              std::to_string(Index) << " -> "
+                                    << "((" << Edge.first->getName() << ","
+                                    << Edge.second->getName() << "),"
+                                    << std::to_string(Size) << ")");
+  }
+}
+
+template<class GraphT, class GT>
+bool CycleEquivalenceAnalysis<GraphT, GT>::isBackedge(EdgeDescriptor &ED) {
+
+  size_t SourceDFSNum = NodeDFSNumMap.at(ED.first);
+  size_t TargetDFSNum = NodeDFSNumMap.at(ED.second);
+
+  return TargetDFSNum < SourceDFSNum;
+}
+
+template<class GraphT, class GT>
+void CycleEquivalenceAnalysis<GraphT, GT>::computeDFSAndSpanningTree(BlockGraph
+                                                                       &Graph) {
+  using ExtType = llvm::df_iterator_default_set<BlockNode *>;
+  using UGT = llvm::GraphTraits<llvm::Undirected<BlockNode *>>;
+  using udf_iterator = llvm::df_iterator<BlockNode *, ExtType, false, UGT>;
+
+  BlockNode *EntryNode = Graph.getEntryNode();
+  auto Begin = udf_iterator::begin(EntryNode);
```

Comment:
`s/Begin/It`

Reply:
Changed this here https://github.com/revng/revng-c/blob/323b04da3a0979143e3207968b19e7142893109e/lib/RestructureCFG/CycleEquivalenceAnalysis.cpp#L106